src/Console/InfectionContainer.php

```diff
@@ -80,143 +80,116 @@
-            return new ParallelProcessRunner($this['dispatcher']);
-        };
-
-        $this['testframework.config.locator'] = function (): TestFrameworkConfigLocator {
-            return new TestFrameworkConfigLocator(
-                $this['infection.config']->getPhpUnitConfigDir() /*[phpunit.dir, phpspec.dir, ...]*/
-            );
-        };
-
-        $this['diff.colorizer'] = static function (): DiffColorizer {
-            return new DiffColorizer();
-        };
-
-        $this['test.file.data.provider.phpunit'] = function (): TestFileDataProvider {
-            return new CachedTestFileDataProvider(
-                new PhpUnitTestFileDataProvider($this['phpunit.junit.file.path'])
-            );
-        };
-
-        $this['version.parser'] = static function (): VersionParser {
-            return new VersionParser();
-        };
-
-        $this['lexer'] = static function (): Lexer {
-            return new Lexer\Emulative([
-                'usedAttributes' => [
-                    'comments', 'startLine', 'endLine', 'startTokenPos', 'endTokenPos', 'startFilePos', 'endFilePos',
-                ],
-            ]);
-        };
-
-        $this['parser'] = function (): Parser {
-            return (new ParserFactory())->create(ParserFactory::PREFER_PHP7, $this['lexer']);
-        };
-
-        $this['pretty.printer'] = static function (): Standard {
-            return new Standard();
-        };
-
-        $this['mutators.config'] = function (): array {
-            $mutatorConfig = $this->getInfectionConfig()->getMutatorsConfiguration();
-
-            return (new MutatorsGenerator($mutatorConfig))->generate();
-        };
-
-        $this['metrics'] = static function (): MetricsCalculator {
-            return new MetricsCalculator();
-        };
-
-        $this['timer'] = static function (): Timer {
-            return new Timer();
-        };
-
-        $this['time.formatter'] = static function (): TimeFormatter {
-            return new TimeFormatter();
-        };
-
-        $this['memory.formatter'] = static function (): MemoryFormatter {
-            return new MemoryFormatter();
-        };
-
-        $this['memory.limit.applier'] = function (): MemoryLimiter {
-            return new MemoryLimiter($this['filesystem'], \php_ini_loaded_file());
-        };
+        return new self([
+            'project.dir' => getcwd(),
+            'filesystem' => static function (): Filesystem {
+                return new Filesystem();
+            },
+            'tmp.dir.creator' => function (): TmpDirectoryCreator {
+                return new TmpDirectoryCreator($this['filesystem']);
+            },
+            'tmp.dir' => function (): string {
+                return $this['tmp.dir.creator']->createAndGet($this->getInfectionConfig()->getTmpDir());
+            },
+            'coverage.dir.phpunit' => function () {
+                return sprintf('%s/%s', $this['coverage.path'], CodeCoverageData::PHP_UNIT_COVERAGE_DIR);
```

Comment:
PHPStan does not happy with using `$this` in a static context

Reply:
And it is right, damn I wanted to do it afterwards; well I'll have to change them now:

```php
static function (Container $container) {
    return sprintf('%s/%s', $container['coverage.path'], CodeCoverageData::PHP_UNIT_COVERAGE_DIR);
}
```